ceph/ceph_admin/apply.py

```diff
@@ -1,85 +1,122 @@
 """
+Represent the ceph orch CLI action 'apply'.
+
 Module to deploy ceph role service(s) using orchestration command
 "ceph orch apply <role> [options] --placement '<placements>' "
 
-this module inherited where service deployed using "apply" operation.
-and also validation using orchestration process list response
+This module inherited where service deployed using "apply" operation and also
+validation using orchestration process list response
 """
-import logging
-from time import sleep
+from typing import Dict
 
-from ceph.ceph import ResourcesNotFoundError
+from ceph.utils import get_nodes_by_id
 
-logger = logging.getLogger(__name__)
+from .common import config_dict_to_string
+from .typing_ import ServiceProtocol
 
 
-class ServiceRemoveFailure(Exception):
+class ServiceApplyFailure(Exception):
```

Comment:
IMO, the name of the exception class could be `OrchApplyFailed`.

Reply:
added `OrchApplyServiceFailure`.